### DISCUSSION TITLE
Logging of healthchecks using a different Log-level

### DISCUSSION THREAD

**Discussion by CommanderStorm** [meilisearch/product#476] [Feedback & Feature Proposal]
I would like to discuss the logging of the `/health`-Endpoint.
If I dont know of a feature, that already exists I am sorry.
Looking at [meilisearch-http/src/main.rs#L18](https://github.com/meilisearch/meilisearch/blob/67b6f4340aaea005471a0ad41c700bde1d88afe9/meilisearch-http/src/main.rs#L18
) and [meilisearch-http/src/routes/mod.rs#L172](https://github.com/meilisearch/meilisearch/blob/67b6f4340aaea005471a0ad41c700bde1d88afe9/meilisearch-http/src/routes/mod.rs#L172) I dont see any related feature.

Currently it is logged like
```
[2022-05-26T23:24:30Z INFO  actix_web::middleware::logger] 10.42.0.1 "GET /health HTTP/1.1" 200 22 "-" "kube-probe/1.22" 0.000070
[2022-05-26T23:24:35Z INFO  actix_web::middleware::logger] 10.42.0.1 "GET /health HTTP/1.1" 200 22 "-" "kube-probe/1.22" 0.000061
[2022-05-26T23:24:40Z INFO  actix_web::middleware::logger] 10.42.0.1 "GET /health HTTP/1.1" 200 22 "-" "kube-probe/1.22" 0.000058
```

I would like the logging on this endpoint to either be:
- Disableable via a configuration option
- being of the DEBUG-loglevel
- being entirely filtered out

This is relevant, since when running meilisearch in kubernetes, these logs are pretty annoying, once healthchecks are enabled.

Is there a way to suppress these health check logs, but still keep all other HTTP request logs, that live in INFO?

If there is not a way currently, would the maintainers be open to a PR implementing such a feature?

Thanks

**Comment by gmourier** [comment]
Hi @CommanderStorm 👋

How do you read the logs? Don't you already have the means to filter out `/health` with a tool? For me, it is the job of the log readers but I could be wrong.

Thanks!

**Reply by CommanderStorm** [reply]
Currently, our monitoring stack revolves around grafana => for logs, we use [Grafana Loki](https://grafana.com/oss/loki/) and have filtered said endpoint in the relevant log-metrics/ alerts

**Comment by CHeavyarms** [comment]
Hello Meilisearch! While searching for an option to turn down or off the logging for the `/health` endpoint, I stumbled on this discussion item, and I believe @CommanderStorm's suggestion is a good one. Namely, the `INFO`-level access logging from that endpoint is quite distracting when used by automated health checks, such as in K8s or Docker Compose/Swarm. Having an option to disable the log entries for successful responses served by endpoint would be quite useful for reducing the noise emanating from Meilisearch, especially in development environments with health checks assisting in initialization and automation.

@gmourier, your questions around log management tooling are likely apt for production deployments, but when working with Meilisearch in a development or pre-production environment, there is often no log filtering readily available.

As of now, I turn the log level to `WARN` during development, but I would encourage the maintainers to consider the options suggested by the original poster!
